File: src/agent_loop/domain/runtime_snapshot.py
"""Runtime snapshot domain model."""

from typing import Literal

from pydantic import BaseModel, ConfigDict

from agent_loop.domain.inbox import Inbox
from agent_loop.domain.thread import Thread


# Domain model
class RuntimeSnapshot(BaseModel):
    """An immutable projection of the complete agent-loop lifecycle."""

    model_config = ConfigDict(frozen=True)

    epoch: int = 0
    revision: int = 0
    status: Literal["idle", "running"] = "idle"
    inbox: Inbox = Inbox()
    thread: Thread = Thread()

File: src/control/tauri-react/src/domain/runtimeSnapshot.ts
import type { Inbox } from "./inbox";
import type { Thread } from "./thread";


// Domain model
export interface RuntimeSnapshot {
  readonly epoch: number;
  readonly revision: number;
  readonly status: "idle" | "running";
  readonly inbox: Inbox;
  readonly thread: Thread;
}

export const EMPTY_RUNTIME_SNAPSHOT: RuntimeSnapshot = {
  epoch: 0,
  revision: 0,
  status: "idle",
  inbox: {
    messages: [],
  },
  thread: {
    messages: [],
  },
};
